File: Low Level Design (LLD)/Problems/06. Splitwise/Java/service/GroupService.java
package service;

import model.Expense;
import model.Group;
import model.User;

import java.util.*;

public class GroupService {
    private Map<String, Group> groups = new HashMap<>();

    public Group createGroup(String groupId, String name) {
        Group group = new Group(groupId, name);
        groups.put(groupId, group);
        return group;
    }

    public void addUserToGroup(String groupId, User user) {
        Group group = groups.get(groupId);
        if (group != null) {
            group.addMember(user);
        }
    }

    public Group getGroup(String groupId) {
        return groups.get(groupId);
    }

    public List<Expense> getGroupExpenses(String groupId) {
        Group group = groups.get(groupId);
        return group != null ? group.getExpenses() : new ArrayList<>();
    }
}

File: Low Level Design (LLD)/Problems/06. Splitwise/Cpp/service/src/GroupService.cpp
#include "../include/GroupService.h"

std::shared_ptr<Group> GroupService::createGroup(const std::string& groupId, const std::string& name) {
    auto group = std::make_shared<Group>(groupId, name);
    groups[groupId] = group;
    return group;
}

void GroupService::addUserToGroup(const std::string& groupId, const std::shared_ptr<User>& user) {
    auto it = groups.find(groupId);
    if (it != groups.end() && it->second) {
        it->second->addMember(user);
    }
}

std::shared_ptr<Group> GroupService::getGroup(const std::string& groupId) const {
    auto it = groups.find(groupId);
    return it != groups.end() ? it->second : nullptr;
}

std::vector<std::shared_ptr<Expense>> GroupService::getGroupExpenses(const std::string& groupId) const {
    auto group = getGroup(groupId);
    return group ? group->getExpenses() : std::vector<std::shared_ptr<Expense>>{};
}
